Integrations/ThreatQ/ThreatQ.py

```diff
@@ -0,0 +1,1248 @@
+    1: 'Spearphish',
+    2: 'Watering Hole',
+    3: 'SQL Injection Attack',
+    4: 'DoS Attack',
+    5: 'Malware',
+    6: 'Watchlist',
+    7: 'Command and Control',
+    8: 'Anonymization',
+    9: 'Exfiltration',
+    10: 'Host Characteristics',
+    11: 'Compromised PKI Certificate',
+    12: 'Login Compromise',
+    13: 'Incident'
+}
+
+TYPE_ID_TO_FILE_TYPE = {
+    1: 'Cuckoo',
+    2: 'CrowdStrike Intelligence',
+    3: 'Early Warning and Indicator Notice (EWIN)',
+    4: 'FireEye Analysis',
+    5: 'FBI FLASH',
+    6: 'Generic Text',
+    7: 'Intelligence Whitepaper',
+    8: 'iSight Report',
+    9: 'iSight ThreatScape Intelligence Report',
+    10: 'IB',
+    11: 'AEC',
+    12: 'Malware Analysis Report',
+    13: 'Malware Initial Findings Report (MFIR)',
+    14: 'Malware Sample',
+    15: 'Packet Capture',
+    16: 'Palo Alto Networks WildFire XML',
+    17: 'PCAP',
+    18: 'PDF',
+    19: 'Private Industry Notification (PIN)',
+    20: 'Spearphish Attachment',
+    21: 'STIX',
+    22: 'ThreatAnalyzer Analysis',
+    23: 'ThreatQ CSV File',
+    24: 'Whitepaper'
+}
+
+HEADERS = {
+    'indicator': ['ID', 'Type', 'Value', 'Description', 'Status',
+                  'TQScore', 'CreatedAt', 'UpdatedAt', 'DBotScore', 'URL'],
+    'adversary': ['ID', 'Name', 'CreatedAt', 'UpdatedAt', 'URL'],
+    'event': ['ID', 'Type', 'Title', 'Description', 'Occurred', 'CreatedAt', 'UpdatedAt', 'URL'],
+    'attachment': ['ID', 'Name', 'Title', 'FileType', 'Size', 'Description', 'MD5', 'CreatedAt', 'UpdatedAt',
+                   'MalwareLocked', 'ContentType', 'URL'],
+    'attrs': ['ID', 'Name', 'Value'],
+    'sources': ['ID', 'Name']
+}
+
+OBJ_DIRECTORY = {
+    'indicator': 'indicators',
+    'adversary': 'adversaries',
+    'event': 'events',
+    'attachment': 'attachments'
+}
+
+RELATED_KEY = {
+    'indicator': 'RelatedIndicator',
+    'adversary': 'RelatedAdversary',
+    'event': 'RelatedEvent'
+}
+
+CONTEXT_PATH = {
+    'indicator': 'ThreatQ.Indicator(val.ID === obj.ID)',
+    'adversary': 'ThreatQ.Adversary(val.ID === obj.ID)',
+    'event': 'ThreatQ.Event(val.ID === obj.ID)',
+    'attachment': 'ThreatQ.File(val.ID === obj.ID)'
+}
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_errors_string_from_bad_request(bad_request_results, status_code):
+    if status_code == 404:
+        return 'Object does not exist.\n'
+
+    # Errors could be retrieved in two forms:
+    # 1. A dictionary of fields and errors list related to the fields, all under 'data' key in the response json object
+    # 2. A list, directly within the response object
+
+    errors_string = 'Errors from service:\n\n'
+
+    # First form
+    errors_dict = bad_request_results.json().get('data', {}).get('errors', {})
+    if errors_dict:
+        for error_num, (key, lst) in enumerate(errors_dict.items(), 1):
+            curr_error_string = '\n'.join(lst) + '\n\n'
+            errors_string += '{0}. In \'{1}\':\n{2}'.format(error_num, key, curr_error_string)
```

Comment:
So it will be like the second form.
```suggestion
            errors_string += 'Error #{0}. In \'{1}\':\n{2}'.format(error_num, key, curr_error_string)
```